File: audiobook/src/main/java/de/ph1b/audiobook/dialog/prefs/AutoRewindDialogFragment.java
package de.ph1b.audiobook.dialog.prefs;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.SeekBar;
import android.widget.TextView;

import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.afollestad.materialdialogs.internal.MDTintHelper;

import javax.inject.Inject;

import butterknife.Bind;
import butterknife.ButterKnife;
import de.ph1b.audiobook.R;
import de.ph1b.audiobook.interfaces.SettingsSetListener;
import de.ph1b.audiobook.persistence.PrefsManager;
import de.ph1b.audiobook.utils.App;


public class AutoRewindDialogFragment extends DialogFragment {
    public static final String TAG = AutoRewindDialogFragment.class.getSimpleName();
    private static final int SEEK_BAR_MIN = 0;
    private static final int SEEK_BAR_MAX = 20;
    @Bind(R.id.textView) TextView textView;
    @Bind(R.id.seekBar) SeekBar seekBar;
    @Inject PrefsManager prefs;
    private SettingsSetListener settingsSetListener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        settingsSetListener = (SettingsSetListener) context;
    }

    private void setText(int progress) {
        String autoRewindSummary = getContext().getResources().getQuantityString(
                R.plurals.pref_auto_rewind_summary, progress, progress);
        textView.setText(autoRewindSummary);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        @SuppressLint("InflateParams") View customView = LayoutInflater.from(getContext()).inflate
                (R.layout.dialog_amount_chooser, null);
        ButterKnife.bind(this, customView);
        App.getComponent().inject(this);


        MDTintHelper.setTint(seekBar, ContextCompat.getColor(getContext(), R.color.accent));
        final int oldRewindAmount = prefs.getAutoRewindAmount();
        seekBar.setMax(SEEK_BAR_MAX - SEEK_BAR_MIN);
        seekBar.setProgress(oldRewindAmount - SEEK_BAR_MIN);
        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                setText(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

        // text
        setText(seekBar.getProgress());

        return new MaterialDialog.Builder(getContext())
                .title(R.string.pref_auto_rewind_title)
                .customView(customView, true)
                .positiveText(R.string.dialog_confirm)
                .negativeText(R.string.dialog_cancel)
                .onPositive(new MaterialDialog.SingleButtonCallback() {
                    @Override
                    public void onClick(@NonNull MaterialDialog materialDialog, @NonNull DialogAction dialogAction) {
                        int newRewindAmount = seekBar.getProgress() + SEEK_BAR_MIN;
                        prefs.setAutoRewindAmount(newRewindAmount);
                        settingsSetListener.onSettingsSet(oldRewindAmount != newRewindAmount);
                    }
                })
                .build();
    }

}

File: audiobook/src/main/kotlin/de/ph1b/audiobook/dialog/prefs/AutoRewindDialogFragment.kt
package de.ph1b.audiobook.dialog.prefs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.LayoutInflater
import android.widget.SeekBar
import android.widget.TextView
import com.afollestad.materialdialogs.MaterialDialog
import de.ph1b.audiobook.R
import de.ph1b.audiobook.injection.App
import de.ph1b.audiobook.interfaces.SettingsSetListener
import de.ph1b.audiobook.persistence.PrefsManager
import javax.inject.Inject


class AutoRewindDialogFragment : DialogFragment() {

    @Inject internal lateinit var prefs: PrefsManager

    private lateinit var textView: TextView
    private lateinit var seekBar: SeekBar
    private lateinit var settingsSetListener: SettingsSetListener

    private val SEEK_BAR_MIN = 0
    private val SEEK_BAR_MAX = 20

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        settingsSetListener = context as SettingsSetListener
    }

    private fun setText(progress: Int) {
        val autoRewindSummary = context.resources.getQuantityString(R.plurals.pref_auto_rewind_summary, progress, progress)
        textView.text = autoRewindSummary
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        App.component().inject(this)

        // view binding
        val customView = LayoutInflater.from(context).inflate(R.layout.dialog_amount_chooser, null)
        textView = customView.findViewById(R.id.textView) as TextView
        seekBar = customView.findViewById(R.id.seekBar) as SeekBar

        val oldRewindAmount = prefs.autoRewindAmount
        seekBar.max = SEEK_BAR_MAX - SEEK_BAR_MIN
        seekBar.progress = oldRewindAmount - SEEK_BAR_MIN
        seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                setText(progress)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {
            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {
            }
        })

        // text
        setText(seekBar.progress)

        return MaterialDialog.Builder(context)
                .title(R.string.pref_auto_rewind_title)
                .customView(customView, true)
                .positiveText(R.string.dialog_confirm)
                .negativeText(R.string.dialog_cancel)
                .onPositive { materialDialog, dialogAction ->
                    val newRewindAmount = seekBar.progress + SEEK_BAR_MIN
                    prefs.autoRewindAmount = newRewindAmount
                    settingsSetListener.onSettingsSet(oldRewindAmount != newRewindAmount)
                }
                .build()
    }

    companion object {
        val TAG = AutoRewindDialogFragment::class.java.simpleName
    }
}
